addons/hr_expense/models/hr_expense.py

```diff
@@ -462,7 +469,8 @@ def _onchange_employee_id(self):
     @api.one
     @api.depends('expense_line_ids', 'expense_line_ids.total_amount')
     def _compute_amount(self):
-        self.total_amount = sum(self.expense_line_ids.mapped('total_amount'))
+        self.total_amount = sum(self.expense_line_ids.mapped('total_amount_company_currency'))
```

Comment:
Please do a separate method as triggers are different. By the way clean the triggers of this one, expense_line_ids is not really required. By the way you broke total_amount meaning.